### PR TITLE
[FLINK-20781] Avoid NPE after SourceOperator is closed.

### PR DESCRIPTION
**Pull request by becketqin**
## What is the purpose of the change
This patch is a band-aid patch that prevent NPE thrown from SourceOperator after it is closed. 

The root cause of the NPE is in the out of order mailbox task shutdown sequence which may result in method invocation on `SourceOperator` after it is closed. A proper fix would be fix the the mailbox shutdown sequence.

## Brief change log
- Avoid NPE after SourceOperator is closed.

## Verifying this change
Ran `UnalignedCheckpointITCase` for over 1500 times in IDE without failure.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (**no**)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (**no**)
  - The serializers: (**no**)
  - The runtime per-record code paths (performance sensitive): (**no**)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn/Mesos, ZooKeeper: (**no**)
  - The S3 file system connector: (**no**)

## Documentation

  - Does this pull request introduce a new feature? (no)
  - If yes, how is the feature documented? (not applicable)

### CODE REVIEW

**Comment by rmetzger** [comment]
I reported the build failure here: https://issues.apache.org/jira/browse/FLINK-17441

**Comment by xintongsong** [comment]
@flinkbot run azure

**Review by xintongsong — APPROVED** [review]
Thanks @becketqin. Changes LGTM.
Let's wait for the green light from AZP before merging.

**Comment by becketqin** [comment]
@flinkbot run azure

**Comment by xintongsong** [comment]
It's weird that the flinkbot command is not working.
I've rebased the latest master and pushed to trigger the ci tests.

**Comment by becketqin** [comment]
@xintongsong Maybe we don't have to wait for this CI test. The CI test from my personal Azure pipeline has passed. https://dev.azure.com/becketqin/apache-flink/_build/results?buildId=131

**Comment by xintongsong** [comment]
Sounds good.
+1 for merging.

**Comment by becketqin** [comment]
Patch merged.
master: 9720e5633cf6b85f78a3b6d6f92ed22a7479658e
release-1.12: 7c7f0839f5d13147c24a9e33fb93a7a6758e844b

**Pull request closed without merging**
